hazelcast-sql/pom.xml

```diff
@@ -228,6 +228,10 @@
                                     <include>com.google.guava:listenablefuture</include>
                                     <include>com.google.guava:failureaccess</include>
                                     <include>org.slf4j:slf4j-api</include>
+                                    <include>net.minidev:json-smart</include>
+                                    <include>com.fasterxml.jackson.core:jackson-core</include>
+                                    <include>com.fasterxml.jackson.core:jackson-databind</include>
+                                    <include>com.fasterxml.jackson.core:jackson-annotations</include>
```

Comment:
Not sure if its a good idea to include `jackson-annotations` in shading and also possibly relocating it, I checked and we don't really use any of the annotations, nor would it matter since it would be strictly internal and references would  be maintained as usual. 

However, it is used by the mapper/json-path library unfortunately, so it is effectively required. Maybe a good alternative would be including it as the only non-shaded dependency.